tests/Moq.Analyzers.Test/Helpers/DoppelgangerTestHelper.cs

```diff
@@ -0,0 +1,133 @@
+#pragma warning disable ECS0200
+
+namespace Moq.Analyzers.Test.Helpers;
+
+/// <summary>
+/// Provides common test data and helpers for doppelganger tests.
+/// Doppelganger tests ensure analyzers don't flag user's custom Mock&lt;T&gt; classes.
+/// </summary>
+internal static class DoppelgangerTestHelper
+{
+    /// <summary>
+    /// Base template for doppelganger tests with a custom Mock&lt;T&gt; class.
+    /// </summary>
+    public const string CustomMockClassTemplate = """
```

Comment:
Types should not have members with visibility set higher than the type's visibility. Find all instances and correct